src/System.Memory/src/System/Buffers/StandardFormat.cs

```diff
@@ -69,33 +71,41 @@ public StandardFormat(char symbol, byte precision = NoPrecision)
         /// </summary>
         public static StandardFormat Parse(ReadOnlySpan<char> format)
         {
-            return Parse(new string(format.ToArray()));
-        }
-
-        /// <summary>
-        /// Converts a classic .NET format string into a StandardFormat
-        /// </summary>
-        public static StandardFormat Parse(string format)
-        {
-            if (string.IsNullOrEmpty(format))
+            if (format.Length == 0)
                 return default;
 
-            char specifier = format[0];
-            byte precision = NoPrecision;
-
-            if (format.Length > 1)
+            char symbol = format[0];
+            byte precision;
+            if (format.Length == 1)
             {
-                if (!byte.TryParse(format.Substring(1), out precision))
-                    throw new FormatException("format");
-
-                if (precision > MaxPrecision)
-                    throw new FormatException("precision");
+                precision = NoPrecision;
+            }
+            else
+            {
+                uint parsedPrecision = 0;
+                for (int srcIndex = 1; srcIndex < format.Length; srcIndex++)
```

Comment:
This allows for parsing of strings like "D000...0009". Is that desired behavior? Should we strictly only except [char][1-9][0-9]?

Reply:
Why not - the regular BCL parsing routines allow leading zeros in their format string.